crates/node/builder/src/builder/mod.rs

```diff
@@ -512,7 +512,18 @@ impl<Node: FullNodeTypes> BuilderContext<Node> {
             "p2p network task",
             |shutdown| {
                 network.run_until_graceful_shutdown(shutdown, |network| {
-                    write_peers_to_file(&network, known_peers_file)
+                    if let Some(peers_file) = known_peers_file {
+                        let known_peers = network.all_peers().collect::<Vec<_>>();
```

Comment:
I would add a `num_known_peers` to `NetworkManager`, which calls `num_known_peers` on `PeersManager` so we avoid this collect